tools/indexer/pkg/database/starrock.go

```diff
@@ -0,0 +1,193 @@
+package database
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/primev/mev-commit/indexer/pkg/beacon"
+	"github.com/primev/mev-commit/indexer/pkg/config"
+	"strings"
+	"time"
+)
+
+type DB struct {
+	Conn *sql.DB
+}
+
+func MustConnect(ctx context.Context, dsn string, maxConns, minConns int) (*DB, error) {
+	conn, err := sql.Open("mysql", dsn)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open StarRocks connection: %w", err)
+	}
+
+	// Configure connection pool
+	conn.SetMaxOpenConns(maxConns)
+	conn.SetMaxIdleConns(minConns)
+	conn.SetConnMaxLifetime(time.Hour)
+	conn.SetConnMaxIdleTime(30 * time.Minute)
+	pingCtx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+	if err := conn.PingContext(pingCtx); err != nil {
+		conn.Close()
+		return nil, fmt.Errorf("StarRocks ping failed: %v", err)
+	}
+
+	return &DB{Conn: conn}, nil
+
+}
+func (db *DB) Close() {
+	db.Conn.Close()
```

Comment:
This doesnt return error?

Also if its the job of the DB to close this, we should not be exposing the conn directly outside.